### PR TITLE
shapes: Arc through

### LINKED ISSUE #206: arc: Add a function `arc-through`
That function should connect two points with an arc touching the third point (see circle-through).

### PR DESCRIPTION
**Pull request by johannes-wolf**
Fixes #206. Thanks to @matthew-e-brown for the initial implementation!

### CODE REVIEW

**Comment by johannes-wolf** [comment]
Rebased. We need to either merge #274 into this or onto 0.2.0 before merging this.

File: src/draw/shapes.typ [hunk 1]
```diff
@@ -304,6 +304,52 @@
...
+
+  let side-on-line(a, b, pt) = {
+    let (x1, y1, ..) = a
+    let (x2, y2, ..) = b
+    let (x,  y, ..)  = pt
+    return (x - x1) * (y2 - y1) - (y - y1) * (x2 - x1)
+  }
+
+  let center-is-left = side-on-line(a, c, center) < 0
+  let b-is-left = side-on-line(a, c, b) < 0
```
Comment (fenjalien) [inline src/draw/shapes.typ:336]: I'm probably missing some context as the function docs are yet to be added but could you add comments explaining this?

**Review by fenjalien — CHANGES_REQUESTED** [review]
What happens when a b c are in a striaght line? I'm guessing a divide by 0 happens somewhere, but could we have the panic message more verbose that it was arc-through? A check might need to be added before `calculate-circle-center-3pt`.

Update the changelog too

File: src/draw/shapes.typ [hunk 2]
```diff
@@ -304,6 +304,52 @@
   },)
 }
```
[the remark this answers is not in the document]
Comment (johannes-wolf) [reply src/draw/shapes.typ:306]: I have added documentation + explanation of the calculation.

**Pull request merged**